config/confighttp/confighttp.go

```diff
@@ -462,15 +472,14 @@ func (hss *ServerConfig) ToServer(_ context.Context, host component.Host, settin
 	otelOpts := []otelhttp.Option{
 		otelhttp.WithTracerProvider(settings.TracerProvider),
 		otelhttp.WithPropagators(otel.GetTextMapPropagator()),
-		otelhttp.WithSpanNameFormatter(func(_ string, r *http.Request) string {
-			return r.URL.Path
+		otelhttp.WithSpanNameFormatter(func(operation string, r *http.Request) string {
+			return strings.TrimPrefix(fmt.Sprintf("%s:%s", operation, r.URL.Path), ":")
```

Comment:
If I understand correctly, this is a shortcut to remove the leading colon in case the prefix is not set. It has the side effect though that if the prefix starts with a colon, it will be erroneously removed. This is probably not very plausible, but I'd rather not have unwanted side effects, even if we think those are theoretical only.

```suggestion
			if len(operation) > 0 {
				return fmt.Sprintf("%s:%s", operation, r.URL.Path)
			}
			return r.URL.Path
```

Reply:
This looks much neater! Thanks!